app/controllers/users_controller.rb

```diff
@@ -1,9 +1,17 @@
 class UsersController < ApplicationController
+  before_action :authenticate_user!
+
   def index
-    @users = User.all
+    user_signed_in?
+    current_user
+    user_session
+    @user = User.all
```

Comment:
- [ ] Your app crashes when I go to the root page (users#index), this is because you're getting all users from the model and storing it in a `@user` variable but in your user index view, you're looping through `@users` instead of `@user`. Because of this error, after login, the user is taken to the users#show page instead of the users#index page which is the root page. Kindly fix this.